agent-inject/agent/annotations.go

```diff
@@ -226,6 +226,11 @@ const (
 	// in the Pod whose volume mounts should be copied onto the Vault Agent init and
 	// sidecar containers. Ignores any Kubernetes service account token mounts.
 	AnnotationAgentCopyVolumeMounts = "vault.hashicorp.com/agent-copy-volume-mounts"
+
+	// AnnotationAgentTemplateConfigExitOnRetryFailure configure whether agent
+	// will exit on template render failures once it has exhausted all its retry
+	// attempts. Defaults to true.
+	AnnotationAgentTemplateConfigExitOnRetryFailure = "vault.hashicorp.com/agent-template-config-exit-on-retry-failure"
```

Comment:
I think we can drop `config` here. This is a really long annotation and annotations can't be larger than 63 characters (this is exactly 63 😄 ): https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set

Reply:
I was a bit worried that this might confuse users into thinking that this is a value on Agent's `template` stanza, but if this is a hard limit then this makes the most sense to do.

Reply:
Maybe we could drop the `agent-` prefix? I feel like that's kind of implied here.
```suggestion
	AnnotationAgentTemplateConfigExitOnRetryFailure = "vault.hashicorp.com/template-config-exit-on-retry-failure"
```

Reply:
Yep, went ahead and did that in 649a051 (and also shortened the variable name).